forc/src/cli/help.rs

```diff
@@ -0,0 +1,298 @@
+            },
+            )*
+            ].concat().into_boxed_str())
+        }
+    }
+}
+
+#[macro_export]
+macro_rules! cli_examples {
+    ($( [ $($description:ident)* => $command:tt $($arg:expr)* ] )*) => {
+            #[cfg(test)]
+            use $crate::serial_test;
+            $(
+            $crate::paste::paste! {
+                #[cfg(test)]
+                #[test]
+                #[serial_test::serial]
+                fn [<$($description:lower _)*:snake example>] () {
+                    let mut proc = std::process::Command::new("cargo");
+                    proc.arg("run");
+                    proc.arg("--bin");
+                    proc.arg(format!("forc-{}", stringify!($command)));
+                    proc.arg("--");
+                    $(
+                        proc.arg($arg);
+                    )*
+
+                    let path = std::path::Path::new("tests");
+                    if path.is_dir() {
+                        proc.current_dir(path);
+                    }
+                    let output = proc.output().expect(stringify!($command));
+                    assert!(output.status.success(), "{}: {:?}", stringify!($($description)*), output);
+                }
+            }
+            )*
+
+        fn help() -> &'static str {
+            Box::leak(format!("EXAMPLES:\n{}", examples()).into_boxed_str())
+        }
+
+        fn print_args(args: Vec<String>) -> String {
+            let mut result = String::new();
+            let mut iter = args.iter().peekable();
+            let mut length = 0;
+            let mut is_multiline = false;
+            let equal = "=".to_owned();
+            loop {
+                let arg = if let Some(arg) = iter.next() {
+                    arg
+                } else {
+                    break;
+                };
+                if length + arg.len() > 70 && iter.peek().is_some() && arg.chars().next() != Some('-') {
+                    // too long, break it into a new line
+                    result.push_str("\\\n     ");
+                    is_multiline = true;
+                    length = 5;
+                }
+                if is_multiline && arg.chars().next() == Some('-') {
+                    // it a multiline arg and the next arg is a flag, put them in their own line
+                    result.push_str("\\\n     ");
+                    length = 5;
+                }
+                result.push_str(arg);
+                length += arg.len();
+                if arg != &equal && iter.peek() != Some(&&equal) {
+                    result.push(' ');
+                    length += 1;
+                }
+            }
+            result
+        }
+
+        fn quote_str(s: &str) -> String {
+            let mut result = String::with_capacity(s.len() + 2); // Initial capacity with room for quotes
+
+            result.push('"');
+            for c in s.chars() {
```

Comment:
```suggestion
            for character in s.chars() {
```

Single letter variables are generally a bad practice / code smell / make things more difficult to maintain and onboard people